sdk/go/common/resource/config/crypt.go

```diff
@@ -100,26 +125,100 @@ func NewBlindingDecrypter() Decrypter {
 
 type blindingCrypter struct{}
 
-func (b blindingCrypter) DecryptValue(ciphertext string) (string, error) {
-	return "[secret]", nil
+func (b blindingCrypter) DecryptValue(_ string) (string, error) {
+	return "[secret]", nil //nolint:goconst
+}
+
+func (b blindingCrypter) BulkDecrypt(ciphertexts []string) (map[string]string, error) {
+	secretMap := map[string]string{}
+	for _, c := range ciphertexts {
+		if _, ok := secretMap[c]; ok {
+			continue
+		}
+		secretMap[c] = "[secret]"
+	}
+	return secretMap, nil
 }
 
 func (b blindingCrypter) EncryptValue(plaintext string) (string, error) {
 	return "[secret]", nil
 }
 
+type CachedCrypter interface {
+	Decrypter
+}
+
+// cachedCrypter is a Decrypter that keeps track if decrypted values, which
+// can be retrieved via SecureValues().
+type cachedCrypter struct {
```

Comment:
So there's another type called `cachingCrypter` in pkg/resource/stack/secrets.go so I wonder if this should be called `cachedDecrypter` to make it clear its just one side of the crypt interface.